src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -1,11 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T extends Machine> {
     /**
      * In your implementations - create 2-3 models, add them to the list and return
      * @return - the list of models
      */
-    List<Object> get();
+    List<? extends Machine> get();
```

Comment:
The return type of the `get` method should be `List<T>` instead of `List<? extends Machine>`. Since `T` is already bounded by `Machine`, using `T` will ensure that the method returns a list of the specific type that implements this interface, which is more type-safe and aligns with the generic type parameter `T`.